From 7fb065effa7972e3c80404729edff9edcdf4be26 Mon Sep 17 00:00:00 2001
From: Cool Lee <cool_lee@aspeedtech.com>
Date: Thu, 20 Nov 2025 09:21:06 +0800
Subject: [PATCH] dts: ast2700-evb: Refine reserved memory

- Introduce `EDAF_128MB` macro to support 128MB or 64MB eDAF DDR mode.
- Adjust reserved memory regions for eDAF, eSPI, PCIe, and SSP based on
  configuration flags (`DUAL_NODE`, `PCIE0_EP`, `PCIE1_EP`).
- Consolidate and reorganize memory addresses to align with updated
  hardware mapping.
- Add comments for clarity on usage in MIX or HW mode (enabled in SPL).

Signed-off-by: Cool Lee <cool_lee@aspeedtech.com>
Change-Id: I3231e7a863b6d1ee642fdc1a85f3ab37bbccaca3
(cherry picked from commit 8cbe21a8361bf9de7817c7c32af33480ecd169ae)
Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 arch/arm/dts/ast2700-evb-reserved-mem.dtsi | 76 +++++++++++++---------
 arch/arm/dts/ast2700-evb.dts               |  1 +
 2 files changed, 46 insertions(+), 31 deletions(-)

diff --git a/arch/arm/dts/ast2700-evb-reserved-mem.dtsi b/arch/arm/dts/ast2700-evb-reserved-mem.dtsi
index d333b18886da..84e5b14e8ee4 100644
--- a/arch/arm/dts/ast2700-evb-reserved-mem.dtsi
+++ b/arch/arm/dts/ast2700-evb-reserved-mem.dtsi
@@ -2,71 +2,85 @@
 
 /* AST2700 reserved memories with no-map property */
 
-#if PCIE0_EP
-bmc_dev0_memory: bmc-dev0-memory@41b800000 {
-	reg = <0x4 0x1b800000 0x0 0x100000>;
+#if DUAL_NODE
+espi1_mmbi_memory: espi1-mmbi-memory@413800000 {
+	reg = <0x4 0x13800000 0x0 0x4000000>;
 	no-map;
 };
 
-xdma_memory0: xdma0 {
-	size = <0x0 0x00100000>;
-	compatible = "shared-dma-pool";
+#if EDAF_128MB
+// This area used for eDAF DDR mode even in MIX mode or HW mode(enabled in SPL)
+edaf1: edaf1-memory@417800000 {
+	reg = <0x4 0x17800000 0x0 0x8000000>;
 	no-map;
 };
-
-pcie0_mmbi0_memory: pcie0-mmbi0-memory@41ba00000 {
-	reg = <0x4 0x1ba00000 0x0 0x200000>;
+#else
+edaf1: edaf1-memory@41b800000 {
+	reg = <0x4 0x1b800000 0x0 0x4000000>;
 	no-map;
 };
 #endif
+#endif
 
-#if PCIE1_EP
-bmc_dev1_memory: bmc-dev1-memory@41b900000 {
-	reg = <0x4 0x1b900000 0x0 0x100000>;
+#if EDAF_128MB
+// This area used for eDAF DDR mode even in MIX mode or HW mode(enabled in SPL)
+edaf0: eadf0-memory@41f800000 {
+	reg = <0x4 0x1f800000 0x0 0x8000000>;
+	no-map;
+};
+#else
+edaf0: eadf0-memory@423800000 {
+	reg = <0x4 0x23800000 0x0 0x4000000>;
 	no-map;
 };
+#endif
 
-xdma_memory1: xdma1 {
-	size = <0x0 0x00100000>;
-	compatible = "shared-dma-pool";
+espi0_mmbi_memory: espi0-mmbi-memory@427800000 {
+	reg = <0x4 0x27800000 0x0 0x4000000>;
 	no-map;
 };
 
-pcie1_mmbi4_memory: pcie1-mmbi4-memory@41bc00000 {
-	reg = <0x4 0x1bc00000 0x0 0x200000>;
+#if PCIE0_EP
+bmc_dev0_memory: bmc-dev0-memory@42b800000 {
+	reg = <0x4 0x2b800000 0x0 0x100000>;
 	no-map;
 };
-#endif
 
-pcie2_mmbi0_memory: pcie2-mmbi0-memory@41be00000 {
-	reg = <0x4 0x1be00000 0x0 0x200000>;
+xdma_memory0: xdma0 {
+	size = <0x0 0x00100000>;
+	compatible = "shared-dma-pool";
 	no-map;
 };
 
-espi0_mmbi_memory: espi0-mmbi-memory@41c000000 {
-	reg = <0x4 0x1c000000 0x0 0x4000000>;
+pcie0_mmbi0_memory: pcie0-mmbi0-memory@42ba00000 {
+	reg = <0x4 0x2ba00000 0x0 0x200000>;
 	no-map;
 };
+#endif
 
-// This area used for eDAF DDR mode even in MIX mode or HW mode(enabled in SPL)
-edaf0: eadf0-memory@420000000 {
-	reg = <0x4 0x20000000 0x0 0x4000000>;
+#if PCIE1_EP
+bmc_dev1_memory: bmc-dev1-memory@42b900000 {
+	reg = <0x4 0x2b900000 0x0 0x100000>;
 	no-map;
 };
 
-#if DUAL_NODE
-espi1_mmbi_memory: espi1-mmbi-memory@424000000 {
-	reg = <0x4 0x24000000 0x0 0x4000000>;
+xdma_memory1: xdma1 {
+	size = <0x0 0x00100000>;
+	compatible = "shared-dma-pool";
 	no-map;
 };
 
-// This area used for eDAF DDR mode even in MIX mode or HW mode(enabled in SPL)
-edaf1: edaf1-memory@428000000 {
-	reg = <0x4 0x28000000 0x0 0x4000000>;
+pcie1_mmbi4_memory: pcie1-mmbi4-memory@42bc00000 {
+	reg = <0x4 0x2bc00000 0x0 0x200000>;
 	no-map;
 };
 #endif
 
+pcie2_mmbi0_memory: pcie2-mmbi0-memory@42be00000 {
+	reg = <0x4 0x2be00000 0x0 0x200000>;
+	no-map;
+};
+
 ssp_memory: ssp-memory@42c000000 {
 	reg = <0x4 0x2c000000 0x0 0x2000000>;
 	no-map;
diff --git a/arch/arm/dts/ast2700-evb.dts b/arch/arm/dts/ast2700-evb.dts
index 3890e0a03a89..9074fc616b49 100644
--- a/arch/arm/dts/ast2700-evb.dts
+++ b/arch/arm/dts/ast2700-evb.dts
@@ -7,6 +7,7 @@
 #include <dt-bindings/net/ti-dp83867.h>
 
 #define DUAL_NODE 0	// 1: DUAL_NODE, 0: SINGLE_NODE
+#define EDAF_128MB 0	// 1: 128MB, 0: 64MB
 #define PCIE0_EP 1	// 1: EP, 0: RC
 #define PCIE1_EP 1	// 1: EP, 0: RC
 #define PCIE2_RC 1	// 1: RC, 0: SGMII
-- 
2.34.1

